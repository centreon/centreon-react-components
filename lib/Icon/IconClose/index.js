"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

require("./close-icon.scss");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var IconClose = function IconClose(_ref) {
  var iconType = _ref.iconType,
      onClick = _ref.onClick;
  return _react2.default.createElement("span", { onClick: onClick, className: "icon-close icon-close-" + iconType });
};

exports.default = IconClose;