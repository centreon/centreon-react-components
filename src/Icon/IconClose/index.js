/* eslint-disable react/jsx-filename-extension */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable react/prop-types */

import React from 'react';
import classnames from 'classnames';
import styles from './close-icon.scss';

const IconClose = ({ iconType, iconPosition, onClick, customStyle }) => (
  <span
    onClick={onClick}
    className={classnames(
      styles['icon-close'],
      { [styles[`icon-close-${iconType}`]]: true },
      styles[iconPosition || ''],
      styles[customStyle || ''],
    )}
  />
);

export default IconClose;
