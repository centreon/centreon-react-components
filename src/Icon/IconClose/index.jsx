/* eslint-disable react/prop-types */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */

import React from 'react';
import classnames from 'classnames';
import styles from './close-icon.scss';

function IconClose({ iconType, iconPosition, onClick, customStyle }) {
  return (
    <span
      onClick={onClick}
      className={classnames(
        styles['icon-close'],
        { [styles[`icon-close-${iconType}`]]: true },
        styles[iconPosition || ''],
        styles[customStyle || ''],
      )}
    />
  );
}

export default IconClose;
