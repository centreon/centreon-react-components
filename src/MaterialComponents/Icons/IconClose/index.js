/* eslint-disable react/jsx-filename-extension */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable react/prop-types */

import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Close from '@material-ui/icons/Close';

const useStyles = makeStyles(() => ({
  root: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  icon: {
    color: '#424242',
    cursor: 'pointer',
    float: 'right',
    fontSize: 32,
    zIndex: 9,
    position: 'absolute',
    right: 11,
    top: 8,
  },
  iconLabel: {
    color: '#009fdf',
    fontSize: 12,
    display: 'inline-block',
    verticalAlign: 'middle',
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    fontWeight: 'bold',
    cursor: 'pointer',
    paddingLeft: 5,
  },
  iconCustomStyle: {
    content: '""',
    position: 'absolute',
    right: 0,
    top: 0,
    width: 54,
    height: 49,
    cursor: 'pointer',
    '&:hover': {
      background: '#707070',
      '& svg': {
        color: '#fff',
      },
    },
  },
}));

function IconClose({ label, customStyle, onClick }) {
  const classes = useStyles();

  return (
    <span onClick={onClick} className={classes.iconCustomStyle}>
      <Close style={customStyle} className={classes.icon} />
      {label && <span className={classes.iconLabel}>{label}</span>}
    </span>
  );
}

export default IconClose;
