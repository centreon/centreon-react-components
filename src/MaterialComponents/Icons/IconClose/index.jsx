/* eslint-disable react/prop-types */

import React from 'react';
import Close from '@material-ui/icons/Close';
import styled from '@emotion/styled';
import MaterialIcon from '../MaterialIcon';

const Label = styled.span(() => ({
  color: '#009fdf',
  fontSize: 12,
  display: 'inline-block',
  verticalAlign: 'middle',
  fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
  fontWeight: 'bold',
  cursor: 'pointer',
  paddingLeft: 5,
}));

const FloatingIcon = styled(MaterialIcon)(() => ({
  content: '""',
  position: 'absolute',
  right: 0,
  top: 0,
  width: 54,
  height: 49,
  '&:hover': {
    background: '#707070',
    '& svg': {
      color: '#fff',
    },
  },
}));

const GreyClose = styled(Close)(() => ({
  color: '#424242',
  cursor: 'pointer',
  float: 'right',
  fontSize: 32,
  zIndex: 9,
  position: 'absolute',
  right: 11,
  top: 8,
}));

function IconClose({ label, ...rest }) {
  return (
    <FloatingIcon {...rest}>
      <GreyClose />
      {label && <Label>{label}</Label>}
    </FloatingIcon>
  );
}

export default IconClose;
