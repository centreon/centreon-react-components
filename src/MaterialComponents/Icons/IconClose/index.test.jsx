/* eslint-disable no-undef */

import React from 'react';
import { create } from 'react-test-renderer';
import IconClose from '.';

describe('IconClose', () => {
  it('renders', () => {
    const wrapper = create(<IconClose />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
