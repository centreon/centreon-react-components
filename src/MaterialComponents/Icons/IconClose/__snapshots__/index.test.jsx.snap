// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconClose renders 1`] = `
.emotion-1 {
  display: inline-block;
  vertical-align: middle;
  height: 24px;
  cursor: pointer;
  color: #707070;
  content: "";
  position: absolute;
  right: 0;
  top: 0;
  width: 54px;
  height: 49px;
}

.emotion-1:hover {
  background: #707070;
}

.emotion-1:hover svg {
  color: #fff;
}

.emotion-0 {
  color: #424242;
  cursor: pointer;
  float: right;
  font-size: 32px;
  z-index: 9;
  position: absolute;
  right: 11px;
  top: 8px;
}

<span
  className="emotion-1"
>
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root emotion-0"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</span>
`;
